test/gui/shared/steps/server_context.py

```diff
@@ -28,12 +29,26 @@ def step(context, user_name, resource_name):
 
 @Then('as "|any|" the file "|any|" should have the content "|any|" in the server')
 def step(context, user_name, file_name, content):
-    text_content = webdav.get_file_content(user_name, file_name)
-    test.compare(
-        text_content,
-        content,
-        f"File '{file_name}' should have content '{content}' but found '{text_content}'",
-    )
+    bytes_from_server = webdav.get_file_content(user_name, file_name)
+
+    if file_name.lower().endswith(".txt"):
+        actual_text = bytes_from_server.decode("utf-8")
+        test.compare(
+            actual_text,
+            content,
+            f"File '{file_name}' should have content '{content}'"
+        )
+    else:
+        # for other types of file, compare local content with server content
+        local_path = get_resource_path(file_name)
+        with open(local_path, "rb") as f:
+            local_bytes = f.read()
+
+        test.compare(
+            bytes_from_server,
+            local_bytes,
+            f"File '{file_name}' should match local copy byte-for-byte"
+        )
```

Comment:
I don't get how that can work.
1. the `file_name` here is the file_name on the server, there might or might not be a file with the same name in the `files-for-upload` folder
2. for comparing with a local file the step definition is super confusing.